akka-docs/rst/scala/http/routing-dsl/directives/basic-directives/extractActorSystem.rst

```diff
@@ -0,0 +1,22 @@
+.. _-extractActorSystem-:
+
+extractActorSystem
+==================
+
+Signature
+---------
+
+.. includecode2:: /../../akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala
+   :snippet: extractActorSystem
+
+Description
+-----------
+
+Extracts the ``ActorSystem`` from the ``RequestContext``, which can be useful when the external API
+in your route needs one.
```

Comment:
please add a `.. warning:: This is only supported when the available Materializer is an ActorMaterializer.` note